script_handle_funk.js

```diff
@@ -0,0 +1,320 @@
+   if (!existingOption) {
+      let newData = document.createElement("option");
+      newData.value = data.toUpperCase();
+      newData.innerText = data.toUpperCase();
+      selectData.appendChild(newData);
+   }
+}
+
+// Function to add a new row to the currency table
+function addDataRowTableCurrency(dataRow) {
+   const tableCurrencyRow = document.querySelector('.table-row-currency');
+
+   let existingRow = Array.from(tableCurrencyRow.querySelectorAll("tr")).find(row => {
+      return row.querySelector("th") && row.querySelector("th").innerText === dataRow;
+   });
+
+   if (!existingRow) {
+      let newRow = document.createElement("tr");
+      let newNameRow = document.createElement("th");
+      newNameRow.scope = "row";
+      newNameRow.innerText = dataRow;
+      newRow.appendChild(newNameRow);
+
+      const numCols = document.querySelectorAll('.table-col-currency th').length;
+      for (let i = 0; i < numCols - 1; i++) { // -1 because first column is the row header
+         newRow.appendChild(document.createElement("td"));
+      }
+
+      tableCurrencyRow.appendChild(newRow);
+   }
+}
+
+// Function to add a new column to the currency table
+function addDataColTableCurrency(dataRow, dataCol, rate) {
+   const tableCurrencyCol = document.querySelector('.table-col-currency');
+   const tableCurrencyRow = document.querySelector('.table-row-currency');
+
+   let existingHeader = Array.from(tableCurrencyCol.querySelectorAll("th")).find(header => header.innerText === dataCol);
+
+   if (!existingHeader) {
+      let newColumnHeader = document.createElement("th");
+      newColumnHeader.scope = "col";
+      newColumnHeader.innerText = dataCol;
+      tableCurrencyCol.appendChild(newColumnHeader);
+
+      const rows = document.querySelectorAll('.table-row-currency tr');
+      rows.forEach(row => {
+         row.appendChild(document.createElement("td"));
+      });  
+   }
+
+   let existingRow = Array.from(tableCurrencyRow.querySelectorAll("tr")).find(row => {
+      return row.querySelector("th") && row.querySelector("th").innerText === dataRow;
+   });
+
+   if (existingRow) {
+      const colIndex = Array.from(tableCurrencyCol.querySelectorAll("th")).findIndex(header => header.innerText === dataCol);
+      if (colIndex > -1) {
+         let cell = existingRow.querySelectorAll("td")[colIndex - 1]; // -1 because row header is the first element
+         cell.innerText = rate;
+      }
+   }
+}
+
+// Function to add a new currency rate
+function addRate() {
+   const currencyFrom = document.querySelector('.currencyName_from').value.toUpperCase().trim();
+   const currencyTo = document.querySelector('.currencyName_to').value.toUpperCase().trim();
+   const rate = document.getElementById('currency_rate').value;
+
+   const existingRate = arrRates.find(ratesObj => ratesObj.base === currencyFrom);
+
+   if (existingRate) {
+      if (existingRate.rates[currencyTo] !== undefined) {
```

Comment:
In JavaScript, you don't need to put the `!== undefined` when doing checks. That is because an undefined value will be automatically treated as false. 

See here: https://developer.mozilla.org/en-US/docs/Glossary/Falsy

Though this is a slightly advanced topic specific to JavaScript, so don't worry.